covfee/loader.py

```diff
@@ -30,82 +32,109 @@ def cli_create_tables():
 class Loader:
     """Translates between different covfee file formats"""
 
-    def __init__(self, project_spec_file=None):
+    _project_spec_file: Path
+    _project_spec_from_python_file: bool
+
+    def __init__(self, project_spec_file: str):
         if not os.path.exists(project_spec_file):
             raise FileNotFoundError("covfee file not found.")
 
-        self.project_spec_file = Path(project_spec_file)
-        self.file_extension = self.project_spec_file.suffix
-        if self.file_extension not in [".py", ".json"]:
-            raise ValueError(f"Unsupported file extension {self.file_extension}")
+        self._project_spec_file = Path(project_spec_file)
+        self._project_spec_from_python_file = self._project_spec_file.suffix == ".py"
+
+        if (
+            not self._project_spec_from_python_file
+            and self._project_spec_file.suffix != ".json"
```

Comment:
We don't support json specs anymore, isn't this a left over from covfee v1?

Reply:
Ok! As discussed, it will be left as is, but json-spec related code will be removed with either in an upcoming iteration or letting Jose do that.